evaluationStage.go

```diff
@@ -17,6 +17,21 @@ const (
 	prefixErrorFormat     string = "Value '%v' cannot be used with the prefix '%v'"
 )
 
+var numberTypes = map[reflect.Kind]struct{}{
+	reflect.Int:     struct{}{},
+	reflect.Int8:    struct{}{},
+	reflect.Int16:   struct{}{},
+	reflect.Int32:   struct{}{},
+	reflect.Int64:   struct{}{},
+	reflect.Uint:    struct{}{},
+	reflect.Uint8:   struct{}{},
+	reflect.Uint16:  struct{}{},
+	reflect.Uint32:  struct{}{},
+	reflect.Uint64:  struct{}{},
+	reflect.Float32: struct{}{},
+	reflect.Float64: struct{}{},
+}
```

Comment:
Entirely to satisfy my own curiosity: why use `struct{}` instead of say `bool`? Convention? Micro-optimization?